test/contracts/core/scenario/twapRebalancingSetScenarios.spec.ts

```diff
@@ -0,0 +1,417 @@
+import { RebalancingSetV3Helper } from '@utils/helpers/rebalancingSetV3Helper';
+import { ValuationHelper } from '@utils/helpers/valuationHelper';
+import { RebalanceTestSetup } from '@utils/helpers/rebalanceTestSetup';
+
+BigNumberSetup.configure();
+ChaiSetup.configure();
+const web3 = getWeb3();
+const CoreMock = artifacts.require('CoreMock');
+const { SetProtocolUtils: SetUtils } = setProtocolUtils;
+const { expect } = chai;
+const blockchain = new Blockchain(web3);
+
+interface RebalancingSetDetails {
+  unitShares: BigNumber;
+  naturalUnit: BigNumber;
+  supply: BigNumber;
+}
+
+interface SetDetails {
+  components: string[];
+  units: BigNumber[];
+  naturalUnit: BigNumber;
+}
+
+interface ComponentSettings {
+  component1Price: BigNumber;
+  component2Price: BigNumber;
+  component1Decimals: number;
+  component2Decimals: number;
+}
+
+interface AuctionSettings {
+  chunkSize: BigNumber;
+  chunkAuctionPeriod: BigNumber;
+  expectedAuctions: number;
+}
+
+interface TWAPScenario {
+  name: string;
+  rebalancingSet: RebalancingSetDetails;
+  currentSet: SetDetails;
+  nextSet: SetDetails;
+  components: ComponentSettings;
+  auction: AuctionSettings;
+}
+
+// Cases to add
+  // cUSDc case to/from WBTC
+
+const scenarios: TWAPScenario[] = [
+  {
+    name: 'ETH 20 MA Set Rebalances 100% WETH to 100% USD',
+    rebalancingSet: {
+      unitShares: new BigNumber(2076796),
+      naturalUnit: new BigNumber(1000000),
+      supply: new BigNumber('20556237207015075000000')
+    },
+    currentSet: {
+      components: ['component1'], // ETH
+      units: [new BigNumber(1000000)],
+      naturalUnit: new BigNumber(1000000),
+    },
+    nextSet: {
+      components: ['component2'], // USDC
+      units: [new BigNumber(307)],
+      naturalUnit: new BigNumber(1000000000000),
+    },
+    components: {
+      component1Price: ether(188),
+      component2Price: ether(1),
+      component1Decimals: 18,
+      component2Decimals: 6,
+    },
+    auction: {
+      chunkSize: ether(2000000),
+      chunkAuctionPeriod: new BigNumber(0),
+      expectedAuctions: 5,
+    }
+  },
+  {
+    name: 'ETHBTC Set Rebalances 100% WETH to 50% WETH/WBTC',
+    rebalancingSet: {
+      unitShares: new BigNumber(799797),
+      naturalUnit: new BigNumber(100000000),
+      supply: new BigNumber('14256596210800000000000')
+    },
+    currentSet: {
+      components: ['component1'], // WETH
+      units: [new BigNumber(1000000)],
+      naturalUnit: new BigNumber(1000000),
+    },
+    nextSet: {
+      components: ['component1', 'component2'], // WETH, WBTC
+      units: [new BigNumber(484880000000000), new BigNumber(1000)],
+      naturalUnit: new BigNumber(10000000000000),
+    },
+    components: {
+      component1Price: ether(188),
+      component2Price: ether(9000),
+      component1Decimals: 18,
+      component2Decimals: 8,
+    },
+    auction: {
+      chunkSize: ether(1000000),
+      chunkAuctionPeriod: new BigNumber(0),
+      expectedAuctions: 2,
```

Comment:
Think it's worthwhile to change the oracle prices in here too between chunk auctions

Reply:
Think there can be an array of chunkAuction objects that define the price AND what aggregate price we expect it to clear at say 50bps or whatever. Since we have this modeled out we could also create an expectation object that we can check against. That object could have unitShares or whatever other data we want to add.

Reply:
If we really want to dig into it we could also include collateral token amounts to check for after each chunkAuction. I believe the model breaks those down as well. That would pretty comprehensively cover the most important part of things, the token flows.